packages/hub/src/sessionflusher.ts

```diff
@@ -17,111 +12,122 @@ type ReleaseHealthAttributes = {
 /**
  * @inheritdoc
  */
-export class SessionFlusher implements SessionFlusherLike {
+export class SessionFlusher {
   public readonly flushTimeout: number = 60;
-  private _pendingAggregates: Record<number, AggregationCounts> = {};
-  private _sessionAttrs: ReleaseHealthAttributes;
-  private _intervalId: ReturnType<typeof setInterval>;
-  private _isEnabled: boolean = true;
-  private _transport: Transport;
+  public pendingAggregates: Record<number, AggregationCounts> = {};
+  public sessionAttrs: ReleaseHealthAttributes;
+  public intervalId: ReturnType<typeof setInterval>;
+  public isEnabled: boolean = true;
+  public transport: Transport;
 
   public constructor(transport: Transport, attrs: ReleaseHealthAttributes) {
-    this._transport = transport;
+    this.transport = transport;
     // Call to setInterval, so that flush is called every 60 seconds
-    this._intervalId = setInterval(() => this.flush(), this.flushTimeout * 1000);
-    this._sessionAttrs = attrs;
+    this.intervalId = setInterval(() => flush(this), this.flushTimeout * 1000);
+    this.sessionAttrs = attrs;
   }
+}
 
-  /** Sends session aggregates to Transport */
-  public sendSessionAggregates(sessionAggregates: SessionAggregates): void {
-    if (!this._transport.sendSession) {
-      logger.warn("Dropping session because custom transport doesn't implement sendSession");
-      return;
-    }
-    void this._transport.sendSession(sessionAggregates).then(null, reason => {
-      logger.error(`Error while sending session: ${reason}`);
-    });
+/** Submits the aggregates request mode sessions to Sentry */
+function sendSessionAggregates(sessionFlusher: SessionFlusher, sessionAggregates: SessionAggregates): void {
+  // TODO: But why to take a Transport when it may be the case that doesn't implement the callback?
+  // The only point of taking `transport` is to call this one function thou.
+  if (!sessionFlusher.transport.sendSession) {
+    logger.warn("Dropping session because custom transport doesn't implement sendSession");
+    return;
   }
```

Comment:
This is really interesting, the only thing the flusher does with `transport` is to call `sendSession` so ... what would be the point of passing a Transport that may not implement such method.

Alternatively, the type system shouldn't even allow me to do so, removing the check from here closer to where probably it would matter.

Since only one thing is being used also, probably have a function type than an object

```ts
type SendSessionHandler = (session: Session | SessionAggregates) => PromiseLike<Response>;
// or

type Transporter = (session: Session | SessionAggregates) => PromiseLike<Response>;
```

Reply:
Doing this actually, let see ... by now I really broke everything 😄